azure/scope/cluster.go

```diff
@@ -455,16 +455,25 @@ func (s *ClusterScope) NodeOutboundLBName() string {
 // OutboundLBName returns the name of the outbound LB.
 func (s *ClusterScope) OutboundLBName(role string) string {
 	if role == infrav1.Node {
+		if s.NodeOutboundLB() == nil {
```

Comment:
re-reading this, shouldn't it return s.NodeOutboundLB().Name instead of ClusterName if s.NodeOutboundLB() is not nil? Aren't we defaulting s.NodeOutboundLB().Name to ClusterName already?

Reply:
we default it for public clusters which will always have a node outbound lb, for private clusters we leave it if nil. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/87d3a087dae4c42d915f9cdc0751f799af430a49/api/v1alpha4/azurecluster_default.go#L201

I can add a check for private cluster if it's more readable.

Reply:
What I mean is shouldn't this func be changed to:

```
func (s *ClusterScope) OutboundLBName(role string) string {
	if role == infrav1.Node {
		if s.NodeOutboundLB() == nil {
		    return ""
		}
		return s.NodeOutboundLB().Name
}
```
?

Otherwise we return the wrong name when the user is using a custom LB name

Reply:
oh I see what you mean. Yes, we should change that although it wouldn't make a huge difference since we ignore and default it to cluster name anyways 😄  https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/87d3a087dae4c42d915f9cdc0751f799af430a49/api/v1alpha4/azurecluster_default.go#L226. IIRC, there was some constraint in cloud provider to have this as cluster name. I'll create a followup issue to address this.